app/controllers/tasks_controller.rb

```diff
@@ -24,13 +34,21 @@ def edit; end
   # POST /tasks
   # POST /tasks.json
   def create
+    puts task_params
     @task = Task.new(task_params)
 
     respond_to do |format|
       if @task.save
-        format.html { redirect_to tasks_path, notice: 'Task was successfully created.' }
-        format.json { render :show, status: :created, location: @task }
+        history_params= {task_id: @task.id, name: @task.name, completed: @task.completed}
+        history=History.new(history_params)
 
+        if history.save
+          format.html { redirect_to tasks_path, notice: 'Task was successfully created.' }
+          format.json { render :show, status: :created, location: @task }
+        else
+          format.html { render :new }
+          format.json { render json: @task.errors, status: :unprocessable_entity }
```

Comment:
I could imagine a scenario where the task saves, but the history does not save. In this case, would there be errors and how would that be communicated to the user?

Reply:
In that case, it would be asked to re-enter the task. However, let me know if you think there is a better way?